bus-mapping/src/evm/stack.rs

```diff
@@ -59,20 +59,15 @@ impl Stack {
     }
 
     /// Generate an new instance of EVM stack given a `Vec<EvmWord>`.
-    pub const fn new() -> Stack {
-        Stack(vec![])
+    pub const fn new(words: Vec<EvmWord>) -> Stack {
```

Comment:
Copy that 😎
I think in that case, `new` and `empty` method would be totally the same.
https://github.com/appliedzkp/zkevm-circuits/pull/150/files/1f2c50a5794d7ef2534544ec63f8e4ace14356ee#diff-bacdb9dcd2047875810d93a4f556759cd3b84f472a86b304b5c9dea36d984ca1R57
Should I remove one of them or just update document?

Reply:
I think you can remove the `empty` method and revert the change on `new` so that it returns an empty Stack.
My reasoning is to follow the convention of other container types like Vec: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.new

And whenever you need to create a stack from a Vector, use the `from_vec` method.  Alternatively we could implement the `From` trait like in Vec: https://doc.rust-lang.org/std/vec/struct.Vec.html#impl-From%3C%26%27_%20%5BT%5D%3E But I think having a `from_vec` is fine.